posts/2025-12-02-project-goals-nov-update.md

```diff
@@ -0,0 +1,239 @@
+* Feedback on the previous experiment was received, with Alex expressing interest in adding POSIX process support to WASI, though this is unlikely in the near future.
+
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/109'><strong>Expose experimental LLVM features for automatic differentiation and GPU offloading</strong></a></div>
+    <div style="flex: initial;"><progress value="3" max="9"></progress>
+</div>
+</div>
+
+* The re-design of the autodiff middle/backend was implemented, reducing the remaining LoC to be upstreamed from 2.5k to 1.1k, split into two PRs ([1](https://github.com/rust-lang/rust/pull/133429) and [2](https://github.com/rust-lang/rust/pull/130060)), which received initial feedback and are expected to land in early December.
+* The preprint of the first paper utilizing `std::autodiff` is [available on Arxiv](https://arxiv.org/abs/2411.17011v1), with code available at [ChemAI-Lab/molpipx](https://github.com/ChemAI-Lab/molpipx/), showcasing significantly faster compilation times in Rust compared to JAX.
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/110'><strong>Extend pubgrub to match cargo&#x27;s dependency resolution</strong></a></div>
+    <div style="flex: initial;"><progress value="2" max="2"></progress>
+</div>
+</div>
+
+* The core data structures of PubGrub have been published as a separate `version-ranges` crate, enabling multiple projects to share this core abstraction and benefit from improvements without waiting for the rest of the project.
+* This is one of many steps required to publish a new `0.3.0` version of the PubGrub crate.
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/112'><strong>Make Rustdoc Search easier to learn</strong></a></div>
+    <div style="flex: initial;"><progress value="2" max="9"></progress>
+</div>
+</div>
+
+* Rustdoc will now show type signatures in the search results page, and the boxing transform behaves more like Hoogle's does.
+* Improvements to matching behavior have been made to fit user expectations.
+
+
+<div style="display: flex;" class="mt2 mb3">
+    <div style="flex: auto;"><a href='https://github.com/rust-lang/rust-project-goals/issues/113'><strong>Next-generation trait solver</strong></a></div>
+    <div style="flex: initial;"><progress value="10" max="13"></progress>
+</div>
+</div>
+
+* The `-Znext-solver=coherence` flag has been stabilized again in PR #130654, indicating potential final stabilization.
+* Refactoring of the current "typing mode" tracking is underway, linked to issue #128, to address issue #106, with an FCP initiated for merging candidates in trait goals, expected to resolve related issues in the new solver.
```

Comment:
these issue numbers are quite misleading :thinking:

The first issue is a types team MCP: https://github.com/rust-lang/types-team/issues/128
The second one is https://github.com/rust-lang/trait-system-refactor-initiative/issues/106